File: module/src/main/java/com/github/kr328/intent/Injector.java
package com.github.kr328.intent;

import android.util.Log;

import com.github.kr328.intent.shared.Constants;

@SuppressWarnings({"unused", "RedundantSuppression"})
public class Injector {
    public static void inject(String process) {
        try {
            switch (process) {
                case "system_server":
                    new SystemInjector().inject();
                    break;
                case "app":
                    new AppInjector().inject();
                    break;
            }
        } catch (Exception e) {
            Log.e(Constants.TAG, "inject " + process + ": " + e, e);
        }
    }
}

File: module/src/main/java/com/github/kr328/intent/Injector.kt
package com.github.kr328.intent

import androidx.annotation.Keep
import com.github.kr328.intent.app.AppInjector
import com.github.kr328.intent.shared.TLog
import com.github.kr328.intent.system.SystemInjector

@Keep
fun main(args: Array<String>) {
    if (args.isEmpty()) return

    try {
        when (args[0]) {
            "system" -> SystemInjector().inject()
            "app" -> AppInjector().inject()
        }
    } catch (e: Throwable) {
        TLog.e("inject ${args[0]}: $e", e)
    }
}
